### PR TITLE
feat: Add AuthenticationFlowBindingOverrides to Keycloak Client

### LINKED ISSUE #116: Add AuthenticationFlowBindingOverrides to Keycloak Client
**Is your feature request related to a problem? Please describe.**
Add the browser and direct grant auth flow settings for clients.

**Describe the solution you'd like**
Add the two fields to the KeycloakClient CRD

**Describe alternatives you've considered**
No alternatives

### PR DESCRIPTION
**Pull request by dougkirkley**
# Pull Request Template

## Description
I needed the ability to use a client specific setup for browser and direct grant flows. This allows overriding those two auth flows on a client basis.

Fixes #116 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Enhancement (non-breaking change which improves an existing feature or documentation)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

## Checklist:
- [X] I have performed a self-review of my code
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Pull Request contains one commit. I squash my commits.

### CODE REVIEW

File: config/samples/v1_v1_keycloakclient.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ spec:
   webUrl: https://argocd.example.com
   defaultClientScopes:
     - argocd_groups
+  authenticationFlowBindingOverrides:
+    browser: "browser"
+    direct_grant: "direct grant"
```
Comment (zmotso) [inline config/samples/v1_v1_keycloakclient.yaml:19]: directGrant

File: controllers/keycloakclient/chain/put_client.go [hunk 2]
```diff
@@ -94,6 +94,16 @@ func (el *PutClient) putKeycloakClient(ctx context.Context, keycloakClient *keyc
...
 
 func (el *PutClient) convertCrToDto(ctx context.Context, keycloakClient *keycloakApi.KeycloakClient, realmName string) (*dto.Client, error) {
+	authFlowOverrides := keycloakClient.Spec.AuthenticationFlowBindingOverrides
+	if authFlowOverrides.Browser != "" || authFlowOverrides.DirectGrant != "" {
+		authFlows, err := el.getAuthFlows(keycloakClient, realmName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get auth flows: %w", err)
+		}
+
+		keycloakClient.Spec.AuthenticationFlowBindingOverrides = *authFlows
```
Comment (zmotso) [inline controllers/keycloakclient/chain/put_client.go:104]: If we generate a new secret in the method `secret, err:= el.getSecret(ctx, keycloakClient)` and then update the keycloakClient CR, we will rewrite the original values `keycloakClient.Spec.AuthenticationFlowBindingOverrides`.

I suggest modifying the function `ConvertSpecToClient (spec *keycloakApi.KeycloakClientSpec, clientSecret, realmName string, authFlowOverride map[string]string)` and passing these values directly.

File: api/v1/keycloakclient_types.go [hunk 3]
```diff
@@ -181,6 +181,10 @@ type KeycloakClientSpec struct {
 	// +nullable
 	// +optional
 	Authorization *Authorization `json:"authorization,omitempty"`
+
+	// AuthenticationFlowBindingOverrides client auth flow overrides
+	// +optional
+	AuthenticationFlowBindingOverrides AuthenticationFlowBindingOverrides `json:"authenticationFlowBindingOverrides,omitempty"`
 }
```
Comment (zmotso) [inline api/v1/keycloakclient_types.go:188]: Add this field to the tests `keycloakclient_controller_integration_test.go`
```
By("Creating a KeycloakClient")
		keycloakClient := &keycloakApi.KeycloakClient{
			ObjectMeta: metav1.ObjectMeta{
				Name:      "test-keycloak-client-with-secret-ref",
				Namespace: ns,
			},
			Spec: keycloakApi.KeycloakClientSpec{
			...
			AuthenticationFlowBindingOverrides ...
```

**Review by zmotso — CHANGES_REQUESTED** [review]
@dougkirkley Thank you for contribution.
Please fix a couple of comments, and we will merge your PR.

**Comment by zmotso** [comment]
@SergK  We can merge this PR.

**Review by SergK — APPROVED** [review]
@dougkirkley thank you for the contribution.
@zmotso thanks for the review

**Pull request merged**